### DISCUSSION TITLE
Some mantine packages are not published

### DISCUSSION THREAD

**Discussion by activenode** [mantinedev/mantine#2720] [Feature requests]
We are trying to fiddle around with the Mantine sources to allow ourselves to get better insights and potentially contribute.

On this path we found that the internal package `@mantine/ds` is published whereas `@mantine/styles-api` and `@mantine/demos` is not. 

However I don't see that it would provide negative impacts doing so. In fact we think that when working with other tooling or trying out packages that having `@mantine/demos` and `@mantine/styles-api`  + `@mantine/labs` published as well it would be nothing but beneficial. 

Is there anything speaking against it?

**Comment by rtivital** [comment]
There is no reason to publish `@mantine/demos` and `@mantine/styles-api` packages, as they are used only for internal development of documentation.

**Reply by activenode** [reply]
I understand it might not be obvious but let me elaborate the reasons to do so:

- From my understanding it wouldn't come with downsides / negative impacts as it would be another package added to the pipeline
- From a consistency point of view then all packages would be available via package manager
- One reason to do so is that it would allow using the `@mantine/demos` and `@mantine/styles-api` package without having to build and clone the whole mantine repo all the time. -> Currently we're fiddling and playing with ideas based on the `demos` and the `styles-api` package and it now requires everyone wanting to do the same to clone the main repo, then run `build:all` etc.
- Having read the aforementioned bullet point I want to point out that having those 2 packages additionally published only comes with benefits without me seeing the downsides at this very moment.


Could you elaborate the things that speak against doing so?

**Reply by rtivital** [reply]
> From a consistency point of view then all packages would be available via package manager

These are not actually packages, but just chunks of code that were extracted from the documentation website to solve build performance issues. Initially, demos and styles api packages were part of the documentation, but gatsby rebuilds everything that is not in node_modules folder in dev mode when changes are made. With current number of demos, it would take ~ 30-40 seconds to rebuild all demos during development. So, the demos are extracted to a separate package, not because they are treated as a separate package.

> Currently we're fiddling and playing with ideas based on the demos and the styles-api package and it now requires everyone wanting to do the same to clone the main repo, then run build:all etc.

You need to `build:all` once, then you can rebuild only packages that had changes, for example `npm run build core demos`. 

> that having those 2 packages additionally published only comes with benefits without me seeing the downsides at this very moment

Publishing these packages will mean that:
- People would expect them to follow semver rules (no breaking change in minor and patches releases) which is usually not the case for these packages as breaking changes are introduced constantly.
- Some people will get confused by these packages and try to use them without any documentation (that happened before).

**Reply by activenode** [reply]
I understand. Okay, no worries, we can publish it with a Pipeline in an internal package manager. Thanks for clarifying.
